cmake/Modules/FindMKL.cmake

```diff
@@ -271,7 +278,19 @@ IF (NOT "${MKL_THREADING}" STREQUAL "SEQ")
   ENDFOREACH(mklrtl)
 ENDIF (NOT "${MKL_THREADING}" STREQUAL "SEQ")
 
-# Second: search for sequential ones
+# Search for parallelized ones with native pthread lib
+FOREACH(mklrtl ${mklrtls} "")
+  FOREACH(mkliface ${mklifaces})
+    FOREACH(mkl64 ${mkl64s} "")
+      IF (NOT MKL_LIBRARIES)
+        CHECK_ALL_LIBRARIES(MKL_LIBRARIES MKL_OPENMP_TYPE MKL_OPENMP_LIBRARY cblas_sgemm
+          "mkl_${mkliface}${mkl64};${mklthread};mkl_core;${mklrtl};pthread;${mkl_m};${mkl_dl}" "")
+      ENDIF (NOT MKL_LIBRARIES)
+    ENDFOREACH(mkl64)
+  ENDFOREACH(mkliface)
+ENDFOREACH(mklrtl)
```

Comment:
I've moved the pthread implementations to take priority over the sequential ones. I'm not sure why has it been that way previously.